Corrected typos and text formatting in different tutorials

rakaar opened the pull request:
Most of the changes made are corrected typos in different tutorial notebooks. Except at these 2 places, I made a bit more different kind of changes
1. In chapter - 1 tutorial - 7,  for showing the padding has been done I have added `in_2d`. Reason it will be more clear to the reader that 20 pixels have been padded on both the dimensions.
2. In chapter - 2 tutorial - 5, I found several places where text formatting was not proper. I have added new cells and changed formatting in many different places. For example, one of the cells which contains python code was written in a markdown cell.

I request the reviewers of this PR to be more vigilant at the above places.

rakaar committed:
Corrected typos and text formatting in different tutorials

Jammy2211 commented:
Thanks!

For the main changes in chapter 1 tutorial 7 / chapter 2 tutorial 5, could you make the changes in the corresponding Python (e.g. .py) scripts of the tutorials, as opposed to the .ipynb notebooks?

Jupyter notebooks are extremely hard to maintain in GitHub as their formatting messes with the change tracker (a quick look at the PR differences will show this). We actually write and maintain the tutorials as .py scripts and then have scripts which automatically convert the .py files to notebooks for the workspace. Changes to the notebooks will be over written when I generate a new set of notebooks from the Python scripts.

It should be fairly obviously looking at the .py files how they are formatted to become notebooks, essentially the marker `# %%` alternates between code cells and markdown cells.

I've updated typos from this PR where possible in the .py files but for the same reason above correct all typos in the .py files where possible!

rakaar committed:
Made typos and corrections in python scripts

rakaar committed:
Added output folder in gitignore

rakaar commented:
@Jammy2211 
I have made changes in the `scripts` folder. Along with changes in the previous commit, some more new changes made are
1. Made changes in chapter 2 tutorial 6. There were some text formatting issues.
2. Added a guideline for `Contribution` in README file.
3.  Added output folder in `gitignore`

Jammy2211 commented:
Amazing, thank you.

I'm tidying up a big release (with a huge refactor of visualization, albeit minor changes to the API). I'll incorporate these changes in the next few days now I'm rebuilding the docs.
